service/matching/matching_engine.go

```diff
@@ -791,46 +792,135 @@ func (e *matchingEngineImpl) DescribeTaskQueue(
 	ctx context.Context,
 	request *matchingservice.DescribeTaskQueueRequest,
 ) (*matchingservice.DescribeTaskQueueResponse, error) {
-	if request.DescRequest.ApiMode == enumspb.DESCRIBE_TASK_QUEUE_MODE_UNSPECIFIED {
-		partition, err := tqid.PartitionFromProto(request.DescRequest.TaskQueue, request.GetNamespaceId(), request.DescRequest.TaskQueueType)
-		if err != nil {
-			return nil, err
-		}
-		pm, err := e.getTaskQueuePartitionManager(ctx, partition, true)
-		if err != nil {
-			return nil, err
+	req := request.GetDescRequest()
+	if req.ApiMode == enumspb.DESCRIBE_TASK_QUEUE_MODE_ENHANCED {
+		// collect internal info
+		physicalInfoByBuildId := make(map[string]map[enumspb.TaskQueueType]*taskqueuespb.PhysicalTaskQueueInfo)
+		for _, taskQueueType := range req.TaskQueueTypes {
+			for i := 0; i < e.config.NumTaskqueueWritePartitions(req.Namespace, req.TaskQueue.Name, taskQueueType); i++ {
+				pm, err := e.getTaskQueuePartitionManager(
+					ctx,
+					tqid.PartitionFromPartitionProto(&taskqueuespb.TaskQueuePartition{
+						TaskQueue:     req.TaskQueue.Name,
+						TaskQueueType: taskQueueType,
+						PartitionId:   &taskqueuespb.TaskQueuePartition_NormalPartitionId{NormalPartitionId: int32(i)},
+					}, request.GetNamespaceId()),
+					true, // todo carly: should this be true?
+				)
+				if err != nil {
+					return nil, err
+				}
+				buildIds, err := e.getBuildIdList(req.GetVersions(), pm)
+				if err != nil {
+					return nil, err
+				}
+				partitionResp, err := pm.Describe(buildIds, req.GetVersions().GetAllActive(), false, req.ReportPollers)
```

Comment:
the partition manager may live in a different host, instead of getting a local partition manager, we should make a DescribeTaskQueuePartition via e.matchingRawClient.
For the interest of not blocking this PR, we can get rid of the for loop and only report the root partition results.
making the DescribeTaskQueuePartition call and aggregating the result can come in a separate PR.

Reply:
ohh ok

Reply:
I didn't know about e.matchingRawClient